### PR TITLE
[RN] Add unlock room function to mobile

### PR DESCRIPTION
**Pull request by zbettenbuk**
Extends the lock button functionality on mobile by unlocking the room if the room is locked and the button is pressed.

### CODE REVIEW

**Comment by saghul** [comment]
can this wait until the toolbox refactor lands? the logic has been moved to a standalone button now.

**Comment by saghul** [comment]
@zbettenbuk I'm hoping today.

File: lang/main.json [hunk 1]
```diff
@@ -82,7 +82,7 @@
         "mute": "Mute / Unmute",
         "videomute": "Start / Stop camera",
         "authenticate": "Authenticate",
-        "lock": "Lock / Unlock room",
+        "lock": "Lock room",
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply lang/main.json]: Because that dialog doesn't do unlock, only lock

**Comment by zbettenbuk** [comment]
@saghul rebased and fixed the conflict. Thanks for notifying!

File: react/features/room-lock/components/RoomLockButton.js [hunk 2]
```diff
@@ -43,7 +50,17 @@ class RoomLockButton extends AbstractButton<Props, *> {
...
      */
     _handleClick() {
-        this.props.dispatch(beginRoomLockRequest());
+        const { _lockUnlockEnabled } = this.props;
+
+        if (_lockUnlockEnabled) {
+            const { dispatch, _locked } = this.props;
+
+            if (_locked) {
+                dispatch(unlockRoom());
```
Comment (saghul) [inline react/features/room-lock/components/RoomLockButton.js]: The new UI we want for this is a new dialog. I know you were not aware of this, but in a nutshell we need a confirmation dialog: "Are you sure you want to unlock the room?" with options of Ok and Cancel.

@lyubomir Do you feel like we should do it in a single go? The dialog can be added later on of course.

**Comment by zbettenbuk** [comment]
Unfortunately the component react-native-prompt we use doesn't seem to support greying out an option (or at least not obvious from the documentation).

Also found a bug around moderator state breaking, ticket submitted: https://jira.atlassian.com/browse/HQL-4236

Addressed the points above, except the confirm dialog, let's wait for Lyubo to decide how to proceed

**Comment by saghul** [comment]
> Also found a bug around moderator state breaking, ticket submitted: https://jira.atlassian.com/browse/HQL-4236

But this is not related to this PR, is it?

File: react/features/base/participants/functions.js [hunk 3]
```diff
@@ -224,3 +225,28 @@ function _getAllParticipants(stateful) {
...
+
+/**
+ * Returns true if the current local participant is a moderator in the
+ * conference.
+ *
+ * @param {Object|Function} stateful - Object or function that can be resolved
+ * to the Redux state.
+ * @returns {boolean}
+ */
+export function isLocalParticipantModerator(stateful: Object | Function) {
```
Comment (saghul) [inline react/features/base/participants/functions.js]: I expected this to be added as a separate commit which also modified the other features to use it, then use it in this very PR. Let's walk this extra mile, please.

File: react/features/room-lock/components/RoomLockButton.js (hunk 2 above)
Comment (lyubomir) [reply react/features/room-lock/components/RoomLockButton.js]: Is the new dialog going to take a lot of time to develop, review, & merge? This PR is supposed to fix a bug which is kind of more urgent than a "feature request" which is a new dialog UI. But if the difference in development time is going to be at most 1 day, I don't mind waiting for the dialog.

**Comment by saghul** [comment]
Jenkins please test this please.

**Comment by lyubomir** [comment]
> I'm not sure if we should land this before that one lands (it would break locking), @lyubomir WDYT?

@saghul, I'm looking at this PR and I don't see how it would break locking. Could you please point me to the breakage?

**Comment by saghul** [comment]
@lyubomir The current code doesn't check for the moderator status. So we try to lock it even if we are not moderators. Because the code for updating our moderator status is flaky (see the PR) our moderator status is 'none', so the locking functionality is disabled.

File: react/features/room-lock/components/RoomLockButton.js [hunk 4]
```diff
@@ -54,7 +62,7 @@ class RoomLockButton extends AbstractButton<Props, *> {
      * @returns {boolean}
      */
     _isDisabled() {
-        return !this.props._conference;
+        return !this.props._localParticipantModerator;
```
Comment (lyubomir) [inline react/features/room-lock/components/RoomLockButton.js]: Ah, the proverbial "it would break locking."
